app/routes/task_routes.py

```diff
@@ -1 +1,133 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+from ..db import db
+from sqlalchemy import asc, desc
+from datetime import datetime
+import requests
+import json
+import os
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+    except KeyError as e:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = {"task": new_task.to_dict()}
+    return response, 201
+
+
+@bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":
+        query = query.order_by(asc(Task.title))
+    if sort_param == "desc":
+        query = query.order_by(desc(Task.title))
+    else:
+        query = query.order_by(Task.id)
+
+    tasks = db.session.scalars(query)
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+
+    return tasks_response
+
+
+@bp.get("/<task_id>")
+def get_one_task_by_id(task_id):
+    task = validate_task(task_id)
+
+    return {"task": task.to_dict()}
+
+# should it be refactored with from_dict??
+
+
+@bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_task(task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+
+    return {"task": task.to_dict()}
+
+
+@bp.patch("/<task_id>/mark_complete")
+def mark_complete(task_id):
+    task = validate_task(task_id)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    # send slack
+    url = "https://slack.com/api/chat.postMessage"
+
+    payload = json.dumps({
+        "channel": "task-notifications",
```

Comment:
Prefer to reference the name of the channel with a constant variable instead of having a string literal in your payload.

Consider using a variable to reference the string that is the name of the channel.
```py
CHANNEL_NAME = "task-notifications"    
{
        "channel": CHANNEL_NAME,
        "text": f"You just completed task: {task.id}:{task.title}!"
}